### PR TITLE
[BEAM-14250] Amended the workaround

### PR DESCRIPTION
**Pull request by KevinGG**
Made row coder work with stringified typehint when a
NamedTupleInMainModule is registered with a RowCoder.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by KevinGG** [comment]
R: @robertwb

Context is: https://github.com/apache/beam/pull/17278
PTAL, thx!

File: sdks/python/apache_beam/coders/typecoders.py [hunk 1]
```diff
@@ -119,15 +119,21 @@ def register_coder(self, typehint_type, typehint_coder_class):
...
+        typehint_type = str(typehint_type)
     self._register_coder_internal(typehint_type, typehint_coder_class)
 
   def get_coder(self, typehint):
     # type: (Any) -> coders.Coder
     if typehint and typehint.__module__ == '__main__':
       # See https://issues.apache.org/jira/browse/BEAM-14250
       # TODO(robertwb): Remove once all runners are portable.
-      typehint = str(typehint)
+      if hasattr(typehint, '__name__'):
```
Comment (robertwb) [inline sdks/python/apache_beam/coders/typecoders.py:133]: `getattr(typehint, '__name__', str(typehint))`

Similarly above.

Comment (KevinGG) [reply sdks/python/apache_beam/coders/typecoders.py:133]: Thanks, will apply it!

**Comment by KevinGG** [comment]
Run PythonFormatter PreCommit

**Pull request merged**